### DISCUSSION TITLE
Revamp extensions catalog by dog-fooding CKAN

### DISCUSSION THREAD

**Discussion by jqnatividad** [ckan/ckan#6163] [Ideas]
https://extensions.ckan.org is currently not maintained.

It uses GitHub pages and the editorial process is quite tedious, with minimum self-service (https://github.com/ckan/extensions.ckan.org).
 
Having maintained it in the past during my stint on the CKAN Steering Group, I created a prototype in 2017, revamping the extensions catalog by [dog-fooding](https://en.wikipedia.org/wiki/Eating_your_own_dog_food) CKAN itself. For several reasons, the prototype was not released.

Coincident with the recent revamp/modernization of CKAN (both the website and the platform itself ), I think its high time the GH pages catalog is replaced with CKAN.

Consider:
* using [ckanext-scheming](https://github.com/ckan/ckanext-scheming) to catalog not only extensions, but existing CKAN instances as well (and how they use the catalogued extensions), by creating several custom metadata schemes.
* using Organizations to catalog the extension developer's affiliation
* using [ckanext-dcat ](https://github.com/ckan/ckanext-dcat#structured-data-and-google-dataset-search-indexing) to register the different metadata schemes with schema.org to increase discoverability (e.g. registering extensions as [SoftwareApplications](https://schema.org/SoftwareApplication), CKAN instances as [Websites](https://schema.org/WebSite), etc.) and to inform the design of the metadata schemas as well
* leveraging CKAN authorization and built-in workflow to allow the community to submit extensions/CKAN instances using a self-service model
* using social components like [ckanext-discourse](https://github.com/OpenGov-OpenData/ckanext-discourse), [ckanext-disqus](https://github.com/ckan/ckanext-disqus), [ckanext-datarequests](https://github.com/conwetlab/ckanext-datarequests), etc. to engender a more active discussion open to more non-technical users (who may not be comfortable with technical fora like GitHub, gitter, etc.).  Doing so also showcases the extensions themselves in action.
* allow more storytelling by using extensions like [ckanext-pages](https://github.com/ckan/ckanext-pages), [ckanext-odata](https://github.com/jqnatividad/ckanext-odata), etc. 

Finally, it will be a great demonstration site for CKAN as far more than an "open data" publishing platform, as is the case with current, skeletal, plain-vanilla [demo site](https://demo.ckan.org ).

**Comment by jqnatividad** [comment]
Perhaps, we can even include some logic in the [extension cookiecutter](https://docs.ckan.org/en/2.9/extensions/tutorial.html) to automagically check/register the extension and check for name collisions.

It can even serve as a way for a CKAN instance to check if an extension has been updated, and if there are any security advisories, acting like the Python Package Index (PyPi) on steroids.

**Reply by anuveyatsu** [reply]
@jqnatividad everything here sounds like a great idea :+1:

**Reply by jqnatividad** [reply]
Thanks @anuveyatsu! Perhaps we can make it a joint effort?

**Reply by anuveyatsu** [reply]
Hi @jqnatividad I missed your last comment here. Note that we also have an awesome list of CKAN related stuff here which isn't really maintained but could be a good/flexible way of doing this sort of work (eg, it is easy to use git/github). It's basically a markdown document: https://github.com/ckan/awesome-ckan
